packages/runtime/container-runtime/src/containerRuntime.ts

```diff
@@ -1673,6 +1676,60 @@ export class ContainerRuntime extends TypedEventEmitter<IContainerRuntimeEvents>
         return this.context.getAbsoluteUrl(relativeUrl);
     }
 
+    /**
+     * Find and reset nodes whose unreferenced state, such as unreferenced time, needs to be reset. These are nodes
+     * that transitioned from `unreferenced -> referenced -> unreferenced` since the last GC run.
+     * @param resultFromCurrentRun - The result of the current GC run.
+     */
+    private resetUnreferencedState(resultFromCurrentRun: IGCResult) {
+        assert(this.gcDataFromLastRun !== undefined, "GC data from last run should be available");
+
+        // Get the list of references that were added since the last GC run.
+        // TODO: The cast to any is done temporarily until the change to IFluidHandleContext in `core-interfaces` is
+        // merged and client is updated to the new `core-interfaces` version.
+        const outboundRoutesSinceLastGC: string[] = (this.IFluidHandleContext as any).decodedHandles;
```

Comment:
Note: The cast to any will be removed once the change in IFluidHandleContext is merged and the client is updated to use the new `core-interfaces` bits.

Reply:
This does not look right, given that this.IFluidHandleContext is of type IFluidHandleContext which does not have decodedHandles. You rely here on concrete class implementation, and there is no guarantee that is what it is.

Reply:
Oh, here it might be fine (I was looking at data store level), but you can avoid it by changing type of IFluidHandleContext() getter to return concrete class, or having internal variable of right type

Reply:
I will update it.